cli/register/servicewait.go

```diff
@@ -164,13 +166,15 @@ func WaitForService(org string, waitService string, waitTimeout int, pattern str
 			for _, ag := range ags {
 				if ag.RunningWorkload.URL == ss.name && (ag.RunningWorkload.Org == ss.org || ss.org == "*") {
 					if ag.AgreementAcceptedTime == 0 {
-						serviceStatus = AgreementFormed
-					} else {
+						serviceStatus = AgreementProposalReceived
+					} else if ag.AgreementFinalizedTime == 0 {
 						serviceStatus = AgreementAccepted
+					} else {
+						serviceStatus = AgreementFinalized
 					}
 				}
 			}
-			if ss.status >= AgreementAccepted {
+			if ss.status >= AgreementFinalized {
```

Comment:
Please still keep `if ss.status >= AgreementAccepted {` because `ServiceCreated` could start before `AgreementFinalized`. They are in different thread.

Reply:
Updated, thanks